doc/releases/changelog-dev.md

```diff
@@ -288,6 +298,14 @@
 * `qml.matrix` is now compatible with qnodes compiled by catalyst.qjit.
   [(#5753)](https://github.com/PennyLaneAI/pennylane/pull/5753)
 
+* `qml.snapshots` raises an error when a measurement other than `qml.state` is requested from `default.qubit.legacy` instead of silently returning the statevector.
```

Comment:
```suggestion
* `qml.snapshots` raises an error when a measurement other than `qml.state` is requested from `default.qubit.legacy` instead of silently returning the statevector.
  [(#5805)](https://github.com/PennyLaneAI/pennylane/pull/5805)
```